packages/wallet-adapter-core-new/src/WalletCoreNew.ts

```diff
@@ -0,0 +1,1020 @@
+
+          const aptos = new Aptos(aptosConfig);
+          const transaction = await aptos.transaction.build.simple({
+            sender: this._account.address.toString(),
+            data: transactionInput.data,
+            options: transactionInput.options,
+          });
+
+          type AptosSignAndSubmitTransactionV1Method = (
+            transaction: AnyRawTransaction
+          ) => Promise<UserResponse<AptosSignAndSubmitTransactionOutput>>;
+
+          const signAndSubmitTransactionMethod = this._wallet.features[
+            "aptos:signAndSubmitTransaction"
+          ]
+            .signAndSubmitTransaction as unknown as AptosSignAndSubmitTransactionV1Method;
+
+          const response = (await signAndSubmitTransactionMethod(
+            transaction
+          )) as UserResponse<AptosSignAndSubmitTransactionOutput>;
+
+          if (response.status === UserResponseStatus.REJECTED) {
+            throw new WalletConnectionError("User has rejected the request")
+              .message;
+          }
+
+          return response.args;
+        }
+
+        const response = await this._wallet.features[
+          "aptos:signAndSubmitTransaction"
+        ].signAndSubmitTransaction({
+          payload: transactionInput.data,
+          gasUnitPrice: transactionInput.options?.gasUnitPrice,
+          maxGasAmount: transactionInput.options?.maxGasAmount,
+        });
+        if (response.status === UserResponseStatus.REJECTED) {
+          throw new WalletConnectionError("User has rejected the request")
+            .message;
+        }
+        return response.args;
+      }
+
+      // If wallet does not support signAndSubmitTransaction
+      // the adapter will sign and submit it for the dapp.
+      const aptosConfig = getAptosConfig(this._network, this._dappConfig);
+
+      const aptos = new Aptos(aptosConfig);
+      const transaction = await aptos.transaction.build.simple({
+        sender: this._account.address,
+        data: transactionInput.data,
+        options: transactionInput.options,
+      });
+
+      const signTransactionResponse = await this.signTransaction(transaction);
+      const response = await this.submitTransaction({
+        transaction,
+        senderAuthenticator:
+          "authenticator" in signTransactionResponse
+            ? signTransactionResponse.authenticator
+            : signTransactionResponse,
+      });
+      return { hash: response.hash };
+    } catch (error: any) {
+      const errMsg = generalizedErrorMessage(error);
+      throw new WalletSignAndSubmitMessageError(errMsg).message;
+    }
+  }
+
+  /**
+   * Signs a transaction
+   *
+   * To support both existing wallet adapter V1 and V2, we support 2 input types
+   *
+   * @param transactionOrPayload AnyRawTransaction - V2 input | Types.TransactionPayload - V1 input
+   * @param options optional. V1 input
+   *
+   * @returns AccountAuthenticator
+   */
+  async signTransaction(
```

Comment:
maybe we need to split it into 2 functions, something like `signRawTransaction` and `signTransactionJson`

see https://aptos-org.slack.com/archives/C07NV7G72NL/p1737654274929199?thread_ts=1737609790.072349&cid=C07NV7G72NL